### ISSUE TITLE
Referencing message inside a message catastrophically breaks the website

### ISSUE THREAD

**Issue by Ksisa** [asyncapi/studio#771]
I was playing around with message extensions (on hosted website https://studio.asyncapi.com/), thinking I could reference the response message inside the message like so

```yaml
components:
  messages:
    TradeMessageServer:
      name: Trade Message (server)
      title: Trade Message (server)
      description: A message containing trade data
      payload:
        title: TradePayloadServer
        type: object
        properties:
          operation:
            type: string
            const: Trade
          data:
            $ref: "#/components/schemas/TradeServer"
        additionalProperties: false
        required:
          - operation
          - data
      x-response: 
        $ref: "#/components/messages/TradeMessageClient" #this breaks it
```

If you enter a valid message into the x-response tag, it completely breaks the website, so much that even ctrl+f5 doesn't work. You need to go into the dev tools and clear the local storage to get it to work again.

![image](https://github.com/asyncapi/studio/assets/53404771/9b96fe46-ef72-4cf7-a63c-064fe71d1a45)

I understand this is likely to be invalid, but I also doubt the website should break in such a catastrophic way.

**Comment by KhudaDad414** [comment]
@Ksisa I have tried to reproduce the error with this file:

```yaml
asyncapi: '2.6.0'
info:
  title: Account Service
  version: 1.0.0
  description: This service is in charge of processing user signups
channels:
  user/signedup:
    subscribe:
      message:
        $ref: '#/components/messages/TradeMessageServer'
components:
  messages:
    TradeMessageServer:
      name: Trade Message (server)
      title: Trade Message (server)
      description: A message containing trade data
      payload:
        title: TradePayloadServer
        type: object
        properties:
          operation:
            type: string
            const: Trade
        additionalProperties: false
        required:
          - operation
          - data
      x-response: 
        $ref: "#/components/messages/TradeMessageClient" #this breaks it
    TradeMessageClient:
      name: Trade Message (server)
      title: Trade Message (server)
      description: A message containing trade data
      payload:
        title: TradePayloadServer
        type: object
        properties:
          operation:
            type: string
            const: Trade
        additionalProperties: false
        required:
          - operation
          - data
```

but it seems to be working fine. can you provide a full example of an AsyncAPI file that causes it to fail?

**Comment by KhudaDad414** [comment]
@Ksisa is the issue resolved?

**Comment by Afler** [comment]
same here

**Comment by Ksisa** [comment]
@KhudaDad414 turns out I made a mistake, it happens when you reference the same message within the message: 
`yaml
asyncapi: '2.6.0'
info:
  title: Account Service
  version: 1.0.0
  description: This service is in charge of processing user signups
channels:
  user/signedup:
    subscribe:
      message:
        $ref: '#/components/messages/TradeMessageServer'
components:
  messages:
    TradeMessageServer:
      name: Trade Message (server)
      title: Trade Message (server)
      description: A message containing trade data
      payload:
        title: TradePayloadServer
        type: object
        properties:
          operation:
            type: string
            const: Trade
        additionalProperties: false
        required:
          - operation
          - data
      x-response: 
        $ref: "#/components/messages/TradeMessageServer" #this breaks it
    TradeMessageClient:
      name: Trade Message (server)
      title: Trade Message (server)
      description: A message containing trade data
      payload:
        title: TradePayloadServer
        type: object
        properties:
          operation:
            type: string
            const: Trade
        additionalProperties: false
        required:
          - operation
          - data`

**Comment by ashmit-coder** [comment]
Hey @KhudaDad414 @Ksisa  as far as I looked into the issue this looks to be a AsyncAPI-react library issue. It seems that the infinite recursive call to message ref is leading to a Stack overflow. If it seems legit I can report it upstream.

**Comment by KhudaDad414** [comment]
@ashmit-coder As far as the studio is concerned, it shouldn't break. maybe show some kind of an error when things like this happen. maybe from parser or asyncapi-react. 🤔

**Comment by ashmit-coder** [comment]
> @ashmit-coder As far as the studio is concerned, it shouldn't break. maybe show some kind of an error when things like this happen. maybe from parser or asyncapi-react. 🤔 

Yeah but I ran the code on https://asyncapi.github.io/asyncapi-react/

As well and that resulted in it breaking as well. So 🤔

**Comment by catosaurusrex2003** [comment]
The issue mentioned here has been fixed in [this PR](https://github.com/asyncapi/asyncapi-react/pull/1108) and is available in `@asyncapi/react-component@2.5.0`.

I noticed that the Studio is currently using `@asyncapi/react-component@^1.2.2`. Is there a specific reason for not upgrading to the latest version?"

**Comment by Shurtu-gal** [comment]
Package is upgraded and proper error msg is shown up.
